vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -109,3 +119,14 @@ def build(self):
             color="grey",
             parent_className="chart_container",
         )
+
+    def _update_graph_theme(self):
+        @callback(
+            Output(self.id, "figure", allow_duplicate=True),
+            Input("theme_selector", "on"),
+            prevent_initial_call="initial_duplicate",
+        )
+        def update_graph_theme(theme_selector_on: bool):
+            patched_figure = Patch()
+            patched_figure["layout"]["template"] = themes.dark if theme_selector_on else themes.light
+            return patched_figure
```

Comment:
Just to remind you that this means that one http request will be sent to the server `per' Graph object on the page (so it may happen that the response is not rendered at the same time for all graphs). 
In the callback implementation we had in the Page model, there was only one request sent to the server, no matter how many Graphs were on the page.

Reply:
@huong-li-nguyen This may be relevant for you in the tabs considerations. So it looks like moving this from page causes and increase in server requests

Reply:
Ah, good point! Is there any time lag visible right now @maxschulz-COL? Then let's move it back, but the logic of the for-loop might have to change then because as soon as we start introducing sub-pages and tiles it will have to look for any Graph objects on page. But don't change the logic now, this should be done in a separate PR when we start introducing other container models

Reply:
It's quite hard to see any lag while the app is hosted locally, so I also recommend returning the callback back to the Page model.

Reply:
Moved this back